### PR TITLE
build(swagger): Fix Swagger build problems

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to USERS.md.

@simster7 this can be merged as soon as green.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -318,7 +318,6 @@ $(VERSION_FILE):
 	touch $(VERSION_FILE)
 
 dist/$(DB).yaml: $(MANIFESTS) $(E2E_MANIFESTS) $(VERSION_FILE)
-	# We additionally disable ALWAYS_OFFLOAD_NODE_STATUS
```
Comment (alexec) [inline Makefile:321]: comment no longer correct

File: Makefile [hunk 2]
```diff
@@ -357,8 +356,8 @@ stop:
 $(GOPATH)/bin/goreman:
 	go get github.com/mattn/goreman
 
-.PHONY: start-aux
-start-aux: $(GOPATH)/bin/goreman
+.PHONY: start
+start: status stop install controller cli executor-image $(GOPATH)/bin/goreman
```
Comment (alexec) [inline Makefile:360]: because `start-aux` now blocks, we do not run `wait` or `env`

File: Makefile [hunk 3]
```diff
@@ -451,7 +448,7 @@ $(GOPATH)/bin/swagger:
 .PHONY: swagger
 swagger: api/openapi-spec/swagger.json
 
-pkg/apis/workflow/v1alpha1/openapi_generated.go:
+pkg/apis/workflow/v1alpha1/openapi_generated.go: $(shell find pkg/apis/workflow/v1alpha1 -type f -not -name openapi_generated.go)
```
Comment (alexec) [inline Makefile:451]: the main bug fix

File: Makefile [hunk 4]
```diff
@@ -461,20 +458,29 @@ pkg/apis/workflow/v1alpha1/openapi_generated.go:
 	  --report-filename pkg/apis/api-rules/violation_exceptions.list
 	$(call restore_go_mod)
 
+dist/kubernetes.swagger.json:
```
Comment (alexec) [inline Makefile:461]: I've reordered this section of the make file to reflect its execution order

File: api/openapi-spec/swagger.json [hunk 5]
```diff
@@ -2196,6 +2196,10 @@
             "type": "string"
           }
         },
+        "depends": {
```
Comment (alexec) [inline api/openapi-spec/swagger.json:2199]: @simster7

File: Makefile [hunk 6]
```diff
@@ -461,20 +458,29 @@ pkg/apis/workflow/v1alpha1/openapi_generated.go:
...
+	./hack/recurl.sh dist/kubernetes.swagger.json https://raw.githubusercontent.com/kubernetes/kubernetes/release-1.15/api/openapi-spec/swagger.json
+
 pkg/apiclient/_.secondary.swagger.json: hack/secondaryswaggergen.go pkg/apis/workflow/v1alpha1/openapi_generated.go dist/kubernetes.swagger.json
 	go run ./hack secondaryswaggergen
 
-dist/swagger.json: $(GOPATH)/bin/swagger $(SWAGGER_FILES) $(MANIFESTS_VERSION_FILE) hack/swaggify.sh
-	swagger mixin -c 684 $(SWAGGER_FILES) | sed 's/VERSION/$(MANIFESTS_VERSION)/' | ./hack/swaggify.sh > dist/swagger.json
+# we always ignore the conflicts, so lets automated figuring out how many there will be and just use that
```
Comment (alexec) [inline Makefile:467]: @sarabala1979 so we never have to change the `-c` again :)

File: Makefile (hunk 2 above)
Comment (simster7) [reply Makefile:360]: Great! This is better imo

File: api/openapi-spec/swagger.json (hunk 5 above)
Comment (simster7) [reply api/openapi-spec/swagger.json:2199]: There it is!

**Comment by simster7** [comment]
FYI: https://github.com/argoproj/argo/pull/3076 Changed some docstrings that were previously not present on the swagger file and therefore aren't reflected on this PR yet.

Please merge with master before merging this PR in order to avoid master having incorrect codegen

**Comment by alexec** [comment]
Synced with master. Can be merged as soon as approved.

**Pull request merged**